openssl: new wrapper targeting OpenSSL 3+

alaviss opened the pull request:
## Summary
* what changed and how?
* why are we changing it?

## Details
* info that couldn't fit into the summary
* relevant details; tricky parts
* anything else

Fixes https://github.com/nim-works/nimskull/issues/1160

---
<!--- Anything before this section break (`---`) will be used as
the commit message when the PR is merged. -->

## Notes for Reviewers
Details to be done after everything is working

<!--
Pull Request(PR) Help

Before Merge Ensure:
* title reads like a short changelog line entry
* code includes tests and is documented
* leave the source better than before, but split out big reformats

See contributor (guide)[https://nim-works.github.io/nimskull/contributing.html]
for details, especially if you're new to this project.

Tips that make PRs easier:
* for big/impactful changes, start with chat/discussions to refine ideas
* refine the pull request message over time; don't have to nail it in one go

alaviss committed:
openssl: new wrapper targeting OpenSSL 3+

alaviss committed:
net: start porting to new openssl wrapper

alaviss committed:
thttpclient_ssl: port to new wrapper

alaviss committed:
openssl: add 1.1.1 support

saem commented on lib/pure/net.nim:
```diff
@@ -826,35 +807,45 @@ when defineSsl:
     ## The handshake must have been completed and the certificate chain must
     ## have been verified successfully or else an empty sequence is returned.
-    ## The chain is ordered from leaf certificate to root certificate.
+    ## The chain is ordered from leaf certificat to root certificate.
```
```suggestion
    ## The chain is ordered from leaf certificate to root certificate.
```

minor typo

alaviss committed:
ci(windows): add pkgconf

alaviss committed:
don't link ssl examples

alaviss committed:
net(windows): re-enable certificate checking for Windows

alaviss committed:
tests: enable SSL tests on Windows

alaviss committed:
thttpclient_ssl: various fixes

- Properly cleanup resources after each test
- Terminate server on error: Previously the server would've kept running
  on error, which might cause later tests to fail
- Server now read data using SSL_read instead of regular socket reads
- Remove usage of threadpool as nothing here requires it

alaviss committed:
openssl: read OPENSSL_LIB_DIR env for library location

alaviss committed:
net(windows): use winstore for CA certificates if available

alaviss committed:
openssl: remove old wrapper

alaviss committed:
topenssl: port to new openssl

alaviss committed:
untestable/tssl: use new wrapper

alaviss committed:
topenssl: support openssl 1.1.1
